File: String/sort-characters-by-frequency.cpp
class Solution {
public:
     
    string frequencySort(string s) {

        unordered_map<char,int>mp;
        vector<string>has(s.size()+1,"");

        set<char>st(s.begin(),s.end());

        if(st.size() == 1)
        {
            return s;
        }

        string res;

        for(char i:s){
            mp[i]++;
        }
        for(auto& it:mp)
        {
            int n = it.second;
            char c = it.first;

            has[n].append(n,c);

        }


        for(int i=s.size()-1;i>0;i--)
        {
            if(!has[i].empty())
            {
                res.append(has[i]);
            }

        }

        return res;






    }
};

File: String/sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        mp = {}
        for char in s:
            if char in mp:
                mp[char] += 1
            else:
                mp[char] = 1
        dt = []
        for key,val in mp.items():
            dt.append([val,key])
        del mp
        dt.sort()
        res = ''
        
        for i in range(len(dt)-1,-1,-1):
            res += (dt[i][0]*dt[i][1])
        del dt
        return res
        
